company-template.el

```diff
@@ -33,6 +33,7 @@
   (let ((keymap (make-sparse-keymap)))
     (define-key keymap [tab] 'company-template-forward-field)
     (define-key keymap (kbd "TAB") 'company-template-forward-field)
+    (define-key keymap (kbd "C-d") 'company-template-clear-field)
```

Comment:
I think we should fall back to delete-char when there is no active overlay. It should be possible to use the default shortcut for deleting a character when editing the variable name. Would be also the same as yasnippets behavior.